### ISSUE TITLE
Trouble with the path variables

### ISSUE THREAD

**Issue by abhi18av** [neon-bindings/neon#217]
Hi @dherman 

I saw your talk on YT and I was inspired to try out something with `rust`. So, the process till the neon based `Hello, World!` is really smooth - no issues there.

But I wanted to try out `llvm-sys crate` to use have some access to the LLVM codegen - but the build fails.

I do notice that the reason is path variables but I'd like to source that from the local pre-compiled llvm files ( copied over from the brew Cellar ).

Here's my directory structure 

```s
lumos_vm git/master*
❯ tree -L 1

.
├── ProgressLog.md
├── ReadMe.md
├── llvm_4.0.0
└── lumos_1

2 directories, 2 files

```

And here's the stacktrace after running `npm install`

```sh


lumos_vm/lumos_1 git/master*
❯ npm install

> lumos_1@0.1.0 install /Users/eklavya/Projects/bitworld/ClojureScript/lumos_vm/lumos_1
> neon build

neon info running cargo
   Compiling lazy_static v0.2.8
   Compiling utf8-ranges v1.0.0
   Compiling libc v0.2.24
   Compiling semver-parser v0.7.0
   Compiling regex-syntax v0.4.1
   Compiling void v1.0.2
   Compiling unreachable v1.0.0
   Compiling thread_local v0.3.4
   Compiling memchr v1.0.1
   Compiling semver v0.6.0
   Compiling aho-corasick v0.6.3
   Compiling regex v0.2.2
   Compiling llvm-sys v40.0.2
error: failed to run custom build command for `llvm-sys v40.0.2`
process didn't exit successfully: `/Users/eklavya/Projects/bitworld/ClojureScript/lumos_vm/lumos_1/native/target/release/build/llvm-sys-632d8c2c2d97ae93/build-script-build` (exit code: 101)
--- stdout
Didn't find usable system-wide LLVM.
LLVM_SYS_40_PREFIX not set, not using precompiled binaries
LLVM_SYS_40_AUTOBUILD not set, will not automatically compile LLVM

--- stderr
thread 'main' panicked at 'Could not find a compatible version of LLVM', /Users/eklavya/.cargo/registry/src/github.com-1ecc6299db9ec823/llvm-sys-40.0.2/build.rs:14
note: Run with `RUST_BACKTRACE=1` for a backtrace.

neon ERR! cargo build failed

Error: cargo build failed
    at Target.<anonymous> (/Users/eklavya/Projects/bitworld/ClojureScript/lumos_vm/lumos_1/node_modules/neon-cli/dist/neon-cli/target.js:124:35)
    at step (/Users/eklavya/Projects/bitworld/ClojureScript/lumos_vm/lumos_1/node_modules/neon-cli/dist/neon-cli/target.js:32:23)
    at Object.next (/Users/eklavya/Projects/bitworld/ClojureScript/lumos_vm/lumos_1/node_modules/neon-cli/dist/neon-cli/target.js:13:53)
    at fulfilled (/Users/eklavya/Projects/bitworld/ClojureScript/lumos_vm/lumos_1/node_modules/neon-cli/dist/neon-cli/target.js:4:58)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7)
npm WARN lumos_1@0.1.0 No repository field.

up to date in 26.123s

lumos_vm/lumos_1 git/master*  27s
❯
```

I'm quite new to systems programming, could you give me pointers?

**Comment by RReverser** [comment]
@abhi18av That's question to llvm-sys, not really to this repo.

But anyway, as [llvm-sys](https://github.com/tari/llvm-sys.rs) documentation says:

> There must be a compatible version of LLVM available. By default llvm-sys will look for llvm-config on PATH to find a system-wide copy of LLVM and use that if it is a compatible version. 

So you need to have LLVM 4.0 (as you are trying to use it) installed on your target machine. Install it via `apt-get` or `brew` or other package manager appropriate to your system and things should work.

**Comment by abhi18av** [comment]
Oh, I see. Hey, feel free to close the issue :)

But I'd like to source the llvm part from within a second directory, as
mentioned in the above tree command.

How to pass that location to the npm build script?

I'd like to find  solution and utomate the process for further experiments.

**Comment by RReverser** [comment]
> Hey, feel free to close the issue :)

I don't have that power in this repo, was just passing by :)

> But I'd like to source the llvm part from within a second directory, as
mentioned in the above tree command.

Personally I wouldn't recommend that unless you want to patch LLVM itself since it can easily take ~10 GB of space and couple of hours to compile it, but if you really don't want to use precompiled binaries, this is also covered in llvm-sys docs (section about specifying path via an environment variable): https://github.com/tari/llvm-sys.rs#llvm-compatibility

**Comment by abhi18av** [comment]
Actually, I'm not trying to mess with the Dragon here. 

I just moved the pre-compiled `llvm` folder which was in the `brew's Cellar` location to the local folder, which I installed earlier.

What I want to do is to create the `neon-node` package, here called `lumos_1` and serve all it's llvm needs via the pre-compiled binaries in the `llvm_4.0.0` folder.

So, I just need to find out how to automate the process of sourcing llvm from this particular folder and passing it to the `neon build` process.

```sh
lumos_vm git/master*
❯ tree -L 1

├── llvm_4.0.0
└── lumos_1

```
